src/System.Numerics.Experimental/Half.cs

```diff
@@ -0,0 +1,357 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Numerics.Experimental
+{
+    // ===================================================================================================
+    // Portions of the code implemented below are based on the 'Berkeley SoftFloat Release 3e' algorithms.
+    // ===================================================================================================
+
+    [Serializable]
+    [StructLayout(LayoutKind.Sequential)]
+    public readonly struct Half : IComparable, IFormattable, IComparable<Half>, IEquatable<Half>
+    {
+        internal const NumberStyles DefaultParseStyle = NumberStyles.Float | NumberStyles.AllowThousands;
+
+        //
+        // Constants for manipulating the internal bit-representation
+        //
+
+        internal const ushort SignMask = 0x8000;
```

Comment:
Seems these fields should either be public or private?

Reply:
If they are private they still take up space in the IL of course, seems a shame since they are "inlined" by the C# compiler. A case where a preprocessor would be nice.

Reply:
Made private. We shouldn't have these be public unless we also plan to expose similar constants on `Single`/`Double`. Such constants which might be useful for certain types of low-level manipulation, but the raw types might also not be the appropriate place to expose them.